src/libPMacc/include/communication/CommunicatorMPI.hpp

```diff
@@ -38,6 +38,35 @@
 namespace PMacc
 {
 
+namespace detail
+{
+    template <unsigned T_DIM>
```

Comment:
it's a bit notorious but we could also do:

leave the general case unimplemented and implement DIM1 explicitly.